pkg/controller/common.go

```diff
@@ -305,6 +313,17 @@ func (c *SharedState) deleteOperationQueue(volumeName string) {
 	lockable.Unlock()
 }
 
+func (c *SharedState) createReplicaRequestsQueue() {
+	c.priorityReplicaRequestsQueue.queue = cache.NewHeap(
```

Comment:
Nice find! Per the [Heap](https://pkg.go.dev/k8s.io/client-go@v0.23.1/tools/cache#Heap) documentation, it is thread-safe. Do we need the mutex at all?

Reply:
Yep seems like we won't, I'll give it a try without the mutex 👍